### PR TITLE
Feature/lm 2679 promise support in chain

### PR DESCRIPTION
**Pull request by anton-locomote**

### CODE REVIEW

File: lib/ChainFind.js [hunk 1]
```diff
@@ -259,6 +242,12 @@ function ChainFind(Model, opts) {
   };
   chain.all = chain.where = chain.find;
 
+  chain.findAsync   = Promise.promisify(chain.find);
+  chain.firstAsync  = Promise.promisify(chain.first);
+  chain.lastAsync   = Promise.promisify(chain.last);
+  chain.runAsync    = Promise.promisify(chain.run);
+  chain.removeAsync = Promise.promisify(chain.remove);
```
Comment (dxg) [inline lib/ChainFind.js]: Howcome we aren't using https://github.com/locomote/node-orm2/blob/41a2d6aa20feffb144cb82226783e39166d24839/lib/Settings.js#L27 ?

File: test/integration/model-find-chain-async.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        .removeAsync()
+        .then(function () {
+          should.equal(hookFired, false);
+          Person.find(function (err, data) {
+            should.equal(data.length, 2);
+            done();
+          });
+        }).catch(function(err) {
+          done(err);
+        });
```
Comment (dxg) [inline test/integration/model-find-chain-async.js]: If you place a `return` before `Person.find`  then you can omit the `.catch` and `done` calls completely.

File: test/integration/model-find-chain-async.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+  });
+
+  after(function () {
+    return db.close();
+  });
+
+  describe(".firstAsync()", function () {
+    before(setup());
+
+    it("should return only the first element", function () {
```
Comment (tttmaximttt) [inline test/integration/model-find-chain-async.js]: missing `done`

File: test/integration/model-find-chain-async.js [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+        .order("-age")
+        .firstAsync()
+        .then(function (JaneDoe) {
+          JaneDoe.name.should.equal("Jane");
+          JaneDoe.surname.should.equal("Doe");
+          JaneDoe.age.should.equal(20);
+        });
+    });
+
+    it("should return null if not found", function () {
```
Comment (tttmaximttt) [inline test/integration/model-find-chain-async.js]: missing `done`

File: test/integration/model-find-chain-async.js [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+        .then(function (Jack) {
+          should.equal(Jack, null);
+        });
+    });
+  });
+
+  describe(".lastAsync()", function () {
+    before(setup());
+
+    it("should return only the last element", function () {
```
Comment (tttmaximttt) [inline test/integration/model-find-chain-async.js]: missing `done` - probably in all async it missing done, review code please

File: lib/ChainFind.js (hunk 1 above)
Comment (anton-locomote) [reply lib/ChainFind.js]: Done in new commit 5f120b6. Please, check

File: test/integration/model-find-chain-async.js (hunk 2 above)
Comment (anton-locomote) [reply test/integration/model-find-chain-async.js]: Done in new commit 5f120b6. Please, check

**Comment by anton-locomote** [comment]
So we found with Max that according to this info: 
http://tobyho.com/2015/12/16/mocha-with-promises/
and Arek's comment mocha support promises. So we don't need done() if return promises.

File: test/integration/model-find-chain-async.js (hunk 5 above)
Comment (dxg) [reply test/integration/model-find-chain-async.js]: `done` is not needed when returning a promise.

**Pull request merged**